cmd/kubeadm/app/phases/addons/proxy/proxy.go

```diff
@@ -51,6 +52,10 @@ func EnsureProxyAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Inte
 		return fmt.Errorf("error when creating kube-proxy service account: %v", err)
 	}
 
+	// validate cfg.API.AdvertiseAddress.
+	if err := validation.ValidateApiserverAdvertiseAddress(cfg.API.AdvertiseAddress); err != nil {
```

Comment:
The addon phase is called by three commands: init, upgrade, and phase addon; all this command executes 
`SetInitDynamicDefaults` (that executes a first validation of the apiserver) before invoking `EnsureProxyAddon` (that execute a second validation)
If possible, please remove the second validation

Reply:
That't true, thanks, sorry didn't notice. : (
Then I guess there don't have other case to validate this flag.

Reply:
@fabriziopandini Besides, need I add unit test of this case? Seems like `masterconfig.go` don't have any unit test.